src/addon_instance.rs

```diff
@@ -100,18 +100,19 @@ impl AddonInstance {
                 self.adapters.insert(msg.data.adapter_id, adapter);
             }
             Message::DeviceAddedNotification(msg) => {
-                match self.adapters.get_mut(&msg.data.adapter_id) {
-                    Some(adapter) => {
-                        adapter.add_device(msg.data.device);
-                    }
-                    None => {
-                        error!("No adapter with id {} found", msg.data.adapter_id)
-                    }
-                }
+                let adapter = self.get_adapter_mut(&msg.data.adapter_id)?;
+                adapter.add_device(msg.data.device);
             }
             _ => {}
         };
 
         Ok(())
     }
+
+    fn get_adapter_mut(&mut self, id: &str) -> Result<&mut Adapter, String> {
+        match self.adapters.get_mut(id) {
+            Some(adapter) => Ok(adapter),
+            None => Err(format!("No adapter with id {} found", id)),
+        }
```

Comment:
```suggestion
        self.adapters
            .get_mut(id)
            .ok_or(format!("No adapter with id {} found", id))
```
There's a fancy stdlib function for that 😅